src/abstractdataframe/sort.jl

```diff
@@ -93,14 +93,15 @@ ordering(col::ColumnIndex, lt::Function, by::Function, rev::Bool, order::Orderin
 
 # DFPerm: defines a permutation on a particular DataFrame, using
 #         a single ordering (O<:Ordering) or a list of column orderings
-#         (O<:AbstractVector{Ordering}), one per DataFrame column
+#         (NTuple of Ordering), one per DataFrame column
 #
 #         If a user only specifies a few columns, the DataFrame
 #         contained in the DFPerm only contains those columns, and
 #         the permutation induced by this ordering is used to
 #         sort the original (presumably larger) DataFrame
 
-struct DFPerm{O<:Union{Ordering, AbstractVector}, T<:Tuple} <: Ordering
+struct DFPerm{O<:Union{Ordering, Tuple{Vararg{Ordering}}},
```

Comment:
The problem is that if you do `sort(df)` you want a single `Ordering` that is reused to avoid excessive compilation. I would assume that `ord isa Ordering` check should be optimized out by the compiler so it should have no performance penalty.

Reply:
I tried simplifying it but I always ended up with compiler allocating.

Reply:
I don't get it. Why would wrapping the `Ordering` in a one-element tuple trigger more compilation or allocations?

Reply:
Ah - one element `Tuple` is not a problem, but in this case we would anyway have to branch if the tuple is one element or matches length of the column vector.

What allocates is if we wanted to have one tuple where each element would hold a tuple consisting of order and column.

Reply:
OK. Anyway always wrapping `Ordering` in a tuple sounds simpler conceptually.

Reply:
I will change it if I can make work it fast.

Reply:
What's the conclusion regarding this?

Reply:
I decided to leave the union (as in the original design) as in the end the logic was simpler (through dispatch).